### ISSUE TITLE
Arrays in `intialValues` field causes issues in `errors` and `touched` stores

### ISSUE THREAD

**Issue by fenjalien** [pablo-abc/felte#17]
Hi, I have a use case where the initial data has an array of fields for data. You could think of it like items on a shopping list. However, when I attempt to implement it the field in the `errors` store becomes `null` instead of an array of nulls and in `touched` its set to `true` instead of an array of `false`s.

I've tried to convert the array to an object with integer indexes as keys, which works. The `errors` and `touched` show the correct object structure but they do not seem to update (`errors` does not get the error messages returned from the validation function and `touched` does not change the fields to true when changed). It also causes some issues with validating.

See example: https://codesandbox.io/s/felte-demo-forked-s2fdq

**Comment by pablo-abc** [comment]
This is something Felte does not support yet but I want to support it. So, the main issues with having multiple fields like this are:

1. I need a way to differentiate wether an array refers to the value of a single field (say checkbox or an input with `multiple`) vs an array of inputs with single values. Felte would have to treat both cases differently.
2. I need a way to let Felte handle this use-case without needing to use the `bind` directive as with your example above. So Felte should somehow be made aware of the _index_ each field occupies in the array.

I've been experimenting with an interface like this:

```javascript
initialValues: {
  items: [
    {
      value: '',
    },
  ]
}
```

That way I could say "if the element of the array is a plain object, take it as a form", which would also allow to actually use `<fieldset>`s as an array.

**Comment by DhyeyMoliya** [comment]
+1 for this feature. There can be arrays of data. For example, array of tags in a blog post.

**Comment by pablo-abc** [comment]
This is currently being worked on in #24 . It seems to be working as expected, I just need to keep up the code coverage. But this feature should be ready sooner rather than later.

**Comment by pablo-abc** [comment]
This feature has been published in felte@0.7.0 @DhyeyMoliya, @fenjalien. The [documentation has been updated as well](https://felte.dev/docs/nested-forms). Feel free to test it and do let me know if anything is missing!
